pkg/auto/release.go

```diff
@@ -0,0 +1,181 @@
+	}
+
+	// Check if we have a release.yaml file in the expected path
+	if exist, err := filesystem.PathExists(d.RootFs, path.RepositoryReleaseYaml); err != nil || !exist {
+		return nil, errors.New("release.yaml not found")
+	}
+
+	r.ReleaseYamlPath = filesystem.GetAbsPath(d.RootFs, path.RepositoryReleaseYaml)
+
+	return r, nil
+}
+
+// PullAsset will execute the release porting for a chart in the repository
+func (r *Release) PullAsset() error {
+	if err := r.git.FetchBranch(r.VR.DevBranch); err != nil {
+		return err
+	}
+
+	if err := r.git.CheckFileExists(r.AssetPath, r.VR.DevBranch); err != nil {
+		return fmt.Errorf("asset version not found in dev branch: %w", err)
+	}
+
+	if err := r.git.CheckoutFile(r.VR.DevBranch, r.AssetPath); err != nil {
+		return err
+	}
+
+	if err := r.git.ResetHEAD(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkAssetReleased(chartVersion string) bool {
+	if _, err := os.Stat(chartVersion); err != nil {
+		return false
+	}
+
+	return true
+}
+
+func mountAssetVersionPath(chart, version string) (assetPath string, assetTgz string) {
+	// example: assets/longhorn/longhorn-100.0.0+up0.0.0.tgz
+	assetTgz = chart + "-" + version + ".tgz"
+	assetPath = "assets/" + chart + "/" + assetTgz
+	return
+}
+
+// UpdateReleaseYaml reads and parse the release.yaml file to a struct, appends the new version and writes it back to the file.
+func (r *Release) UpdateReleaseYaml() error {
+	var releaseVersions ReleaseVersions
+
+	rvBytes, err := os.ReadFile(r.ReleaseYamlPath)
+	if err != nil {
+		return err
+	}
+
+	if err := yaml.Unmarshal(rvBytes, &releaseVersions); err != nil {
+		return err
+	}
+
+	// Append new version and remove duplicates if any
+	releaseVersions[r.Chart] = append(releaseVersions[r.Chart], r.ChartVersion)
+	releaseVersions[r.Chart] = removeDuplicates(releaseVersions[r.Chart])
+
+	// Marshal the updated releaseVersions back to YAML
+	updatedYAML, err := yaml.Marshal(releaseVersions)
+	if err != nil {
+		return err
+	}
+
+	// Post-process YAML to adjust indentation for list items
+	updatedYAMLIndented := enforceYamlStandard(updatedYAML)
+
+	// Write the updated YAML back to the file
+	if err := os.WriteFile(r.ReleaseYamlPath, updatedYAMLIndented, 0644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// removeDuplicates takes a slice of strings and returns a new slice with duplicates removed.
+func removeDuplicates(slice []string) []string {
+	seen := make(map[string]struct{}) // map to keep track of seen strings
+	var result []string               // slice to hold the results
+
+	for _, val := range slice {
+		if _, ok := seen[val]; !ok {
+			seen[val] = struct{}{}       // mark string as seen
+			result = append(result, val) // append to result if not seen before
+		}
+	}
+
+	return result
+}
+
+// enforceYamlStandard adds indentation to list items in a YAML string and a new line at the end of the file.
+func enforceYamlStandard(yamlBytes []byte) []byte {
```

Comment:
Is there no library for formatting or validating?

Reply:
This was removed. 
I am using `"gopkg.in/yaml.v3"` now.